packages/common/src/repository/activeSubmissionRepository.ts

```diff
@@ -85,15 +85,40 @@ const repository = (dependencies: Dependencies) => {
 				throw new ServiceUnavailable();
 			}
 		},
+
+		/**
+		 * Finds a Submission by ID
+		 * @param {number} submissionId Submission ID
+		 * @returns The Submission found
+		 */
+		getSubmissionById: async (submissionId: number): Promise<Submission | undefined> => {
+			try {
+				return await db.query.submissions.findFirst({
+					where: and(eq(submissions.id, submissionId)),
+				});
+			} catch (error) {
+				logger.error(LOG_MODULE, `Failed getting Submission with id '${submissionId}'`, error);
+				throw new ServiceUnavailable();
+			}
+		},
 		/**
 		 * Update a Submission record in database
 		 * @param {number} submissionId Submission ID to update
 		 * @param {any} newData Set fields to update
 		 * @returns An updated record
 		 */
 		update: async (submissionId: number, newData: Partial<Submission>): Promise<Submission> => {
-			const updated = await db.update(submissions).set(newData).where(eq(submissions.id, submissionId)).returning();
-			return updated[0];
+			try {
+				const updated = await db
+					.update(submissions)
+					.set({ ...newData, updatedAt: new Date() })
+					.where(eq(submissions.id, submissionId))
+					.returning();
+				return updated[0];
```

Comment:
will there be more than one element in this array?

Reply:
https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess

Reply:
the db client returns a Submission[], I agree we should consider if something unexpected happens and array is empty. should we throw an error as you suggested? or should we add on the return type `Promise<Submission | undefined>` and let who's calling this function handle the undefined situation?

Reply:
added a validation to return the value only if it's an array with values